java/lance-jni/src/blocking_dataset.rs

```diff
@@ -0,0 +1,36 @@
+use arrow::ffi_stream::ArrowArrayStreamReader;
```

Comment:
Lets add apache license to headers :) 

`cargo clippy` will pick this rule up. Lets make github action to run on this PR?

Reply:
Good call, I will add the github actions for both java and rust-jni projects